### ISSUE TITLE
A half-gray half-teal (and maybe half-gray half-green?) build that is gray but with the basic generic QOL tweaks 

### ISSUE THREAD

**Issue by NintendoManiac64** [libertyernie/Nintendont#17]
I was thinking about what is effectively a gray build but with some of the basic quality-of-life improvements, specifically...

- the dpad mappings for various games (Four Swords Adventure, Mario Party 4, Smash Bros. Melee, could even add Mario Kart Double Dash without all the other custom controls for that game)
- not needing to fully "click" the analog shoulder button on the non-pro CC for various games (Midway Arcade Treasures 3, Mario Party 4, Spy Hunter, etc)
- having CC-pro_ZR = GC_R, CC-pro_ZL = GC_L, CC-pro_R = GC_Z, and CC-pro_L = GC_Y
- always transmitting a 255/255 GC_R/L-analog-press when doing a GC_R/L-digital-press (perhaps "gray" already has this?)

My rule of thumb was that, if it's a QOL tweak that applies to at least 3 different games, then it probably should be included in this "half-gray half-teal" build.

In terms of actual color name choices to pick from, perhaps refer to:

- https://creativebooster.net/blogs/colors/shades-of-teal-color

The ones that seem to be a "half-gray half-teal" (not including "gray teal" and "teal blue" for obvious reasons) are:

- Bayou
- Cadet
- Dull
- Dusty
- Hooker
- Viridian

And honestly, of those, viridian might be the only one that could even work since most of the others are existing adjectives (dull, dusty) or existing nouns (bayou, cadet, hooker), and viridian is obviously notable to Nintendo players due to Viridian City.

*However*, if one were to make an Xbox 360-enabled version of this, then "viridian" would actually be a much more appropriate name for that, in which case my second choice for a half-gray half-teal build would probably be "cadet".

...but at the same time, there are more well-known colors like "mint" that could be an even better choice for "half-gray half-green" than even "viridian", so I don't know—I'm not a professional digital artist.

Furthermore, a color like "mint" might actually be more appropriate for a variation of grey that has no changes other than adding Xbox 360 controller support.

**Comment by libertyernie** [comment]
My concern is that I would have to repeat the custom controller code, since the D-pad mappings and analog trigger click adjustments are specific to certain games. Does this offer advantages over the teal build? The only one I can think of right now is that it could let us keep both Xbox 360 controller support and Nunchuk support in the same build, but I'm not sure I want to maintain two sets of game-specific checks in the source code (or add other checks inside of those checks for which type of build it is, since that would make the code harder for me to read).

The advantage of the gray build, in my eyes, is that every visual instruction in a game to press a certain button maps directly to a button on the controller with that name, but perhaps I'll add a compiler setting to swap the front and back shoulder button mappings in the gray build - the code specific to the gray build is currently very small, maybe even smaller than the Nintendont default shoulder button code that it replaces.

The gray build does include the 255/255 thing, that's what fixed #13.

**Comment by NintendoManiac64** [comment]
So basically the only possible additions would the "always use 'big' shoulder buttons" and the 255/255 thing, the former of which already occurs in stock Nintendont and the latter of which *could* be implemented in stock Nintendont (which this spurred me on to finally create: https://github.com/FIX94/Nintendont/issues/1178).

Of course, I suppose there's the "small L"=Y thing as well...

**Comment by libertyernie** [comment]
Yeah, maybe just a variant of the gray build that lets you swap the buttons if it's not an original Classic Controller. I might call it black (since Classic Controller Pro comes in black and the original never did)

**Comment by libertyernie** [comment]
Added black build: https://github.com/libertyernie/Nintendont/releases/tag/v20240330

This one maps L and R to the bigger buttons and Y and Z to the smaller ones. Also applies to the GamePad.
